discord/message.py

```diff
@@ -232,6 +232,24 @@ def __init__(self, state, **kwargs):
         self.guild_id = utils._get_as_snowflake(kwargs, 'guild_id')
         self._state = state
 
+    @classmethod
+    def from_message(cls, message):
+        """Creates a :class:`MessageReference` from an existing :class:`Message`.
+
+        .. versionadded:: 1.6
+
+        Parameters
+        ----------
+        message: :class:`Message`
+            The message to be converted into a reference.
+
+        Returns
+        -------
+        :class:`MessageReference`
+            A reference to the message.
+        """
+        return cls(message._state, message_id=message.id, channel_id=message.channel.id, guild_id=message.guild and message.guild.id)
```

Comment:
Prefer `getattr(message.guild, 'id', None)` over  `message.guild and message.guild.id`

Reply:
Benchmarking tests concur, `getattr` is significantly faster.